include/alpaka/event/EventGeneric.hpp

```diff
@@ -0,0 +1,508 @@
+            -> bool
+            {
+                return (m_spEventImpl == rhs.m_spEventImpl);
+            }
+            //-----------------------------------------------------------------------------
+            auto operator!=(EventGeneric<TDev> const & rhs) const
+            -> bool
+            {
+                return !((*this) == rhs);
+            }
+            //-----------------------------------------------------------------------------
+            ~EventGeneric() = default;
+
+        public:
+            std::shared_ptr<generic::detail::EventGenericImpl<TDev>> m_spEventImpl;
+        };
+    }
+
+    namespace dev
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CPU device event device get trait specialization.
+            template<typename TDev>
+            struct GetDev<
+                event::EventGeneric<TDev>>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getDev(
+                    event::EventGeneric<TDev> const & event)
+                -> TDev
+                {
+                    return event.m_spEventImpl->m_dev;
+                }
+            };
+        }
+    }
+    namespace event
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CPU device event test trait specialization.
+            template<typename TDev>
+            struct Test<
+                event::EventGeneric<TDev>>
+            {
+                //-----------------------------------------------------------------------------
+                //! \return If the event is not waiting within a queue (not enqueued or already handled).
+                ALPAKA_FN_HOST static auto test(
+                    event::EventGeneric<TDev> const & event)
+                -> bool
+                {
+                    std::lock_guard<std::mutex> lk(event.m_spEventImpl->m_mutex);
+
+                    return event.m_spEventImpl->isReady();
+                }
+            };
+        }
+    }
+    namespace queue
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CPU non-blocking device queue enqueue trait specialization.
+            template<
+                typename TDev>
+            struct Enqueue<
+                queue::generic::detail::QueueGenericNonBlockingImpl<TDev>,
+                event::EventGeneric<TDev>>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto enqueue(
+#if !(BOOST_COMP_CLANG_CUDA && BOOST_ARCH_PTX)
+                    queue::generic::detail::QueueGenericNonBlockingImpl<TDev> & queueImpl,
```

Comment:
please use `alpaka::ignore_unused(queueImpl)` in the function body.

```
#if BOOST_COMP_CLANG_CUDA && BOOST_ARCH_PTX
    alpaka::ignore_unused(queueImpl);
#endif
```

Reply:
Could be that this code is derived from the original implementation but we should change it directly in this PR.

Reply:
indeed